Sources/SwiftDocC/Model/Rendering/References/CallToActionReference.swift

```diff
@@ -0,0 +1,53 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+/// A specialized ``DownloadReference`` used for ``CallToAction`` directives.
+///
+/// `@CallToAction` directives can link either to a local file or to a URL, whether relative or
+/// absolute. Directives that use the `file` argument will create a ``DownloadReference`` and copy
+/// the file from the catalog into the resulting archive.
+///
+/// A `CallToActionReference` is intended to encode to Render JSON compatible with a
+/// ``DownloadReference``, but with the `url` set to the text given in the `@CallToAction`'s `url`
+/// argument.
+public struct CallToActionReference: RenderReference, URLReference {
```

Comment:
Should this be an `ExternalLocationReference` or something along those lines? `CallToActionReference` feels wrong since sometimes we _do_ use `DownloadReference` for `CallToAction` directives.

Reply:
That could work. I wrote the name before i realized that i wanted to keep using `DownloadReference` for `file` arguments, and i never revisited it. 😅